pkg/config/config.go

```diff
@@ -182,3 +184,48 @@ func findIsoSize(vmi *v1.VirtualMachineInstance, volume *v1.Volume, emptyIso boo
 	}
 	return 0, nil
 }
+
+type volumeInfo interface {
+	isValidType(*v1.Volume) bool
+	getSourcePath(*v1.Volume) string
+	getIsoPath(*v1.Volume) string
+	getLabel(*v1.Volume) string
+}
+
+func createIsoDisksForConfigVolumes(vmi *v1.VirtualMachineInstance, emptyIso bool, info volumeInfo) error {
```

Comment:
there are a couple of things that I don't like of that api (I think it was my first approach):
- it's not obvious (so error-prone) that the caller _must_  pass volumes already filtered to createIsoDisksForConfigVolumes()
- sourceAndTargetPath() returning a tuple I think is error-prone, you need to check which one go first/second the source/target.

I like that you can pass a lambda expression, tho'

Reply:
Btw, feel free to remove the hold (/hold cancel).

>it's not obvious (so error-prone) that the caller must pass volumes already filtered to createIsoDisksForConfigVolumes()

>sourceAndTargetPath() returning a tuple I think is error-prone, you need to check which one go first/second the source/target.

Sure but the caller just shot their own foot because they did not read the manual for the gun... In other words, it is as easy as to provide the wrong implementation to your defined interface.

I would at least suggest considering the change of `vmi` to `volumes`(not filtered) to prevent some crazy ideas and to help keep the focus of the function

Reply:
> 
> Sure but the caller just shot their own foot because they did not read the manual for the gun... 

:)

> In other words, it is as easy as to provide the wrong implementation to your defined interface.

Sure, but I think it's a bit more obvious

> I would at least suggest considering the change of `vmi` to `volumes`(not filtered) to prevent some crazy ideas and to help keep the focus of the function

In that case, we also need to pass the `vmi.Spec.Domain.Devices.Disks`and the `vmi.Status.VolumeStatus`

Reply:
Disregard my comment :)